middleware/index.ts

```diff
@@ -158,18 +158,21 @@ function createOrg(logger: Logger, reqContext: any, accessToken?: string): Org {
 /**
  * Construct Context from function payload.
  *
+ * @param id                   -- invocation id
+ * @param logger               -- logger
+ * @param secrets              -- secrets convenience class
  * @param reqContext           -- reqContext from the request, contains salesforce stuff (user reqContext, etc)
  * @param accessToken          -- accessToken for function org access, if provided
  * @param functionInvocationId -- FunctionInvocationRequest ID, if applicable
  * @return context
  */
-function createContext(id: string, logger: Logger, reqContext?: any, accessToken?: string, functionInvocationId?: string): Context {
+function createContext(id: string, logger: Logger, secrets: Secrets, reqContext?: any,
+                       accessToken?: string, functionInvocationId?: string): Context {
     if (typeof reqContext === 'string') {
         reqContext = JSON.parse(reqContext);
     }
 
     const org = reqContext ? createOrg(logger, reqContext!, accessToken) : undefined;
-    const secrets = createSecrets(logger);
```

Comment:
Moved to initial section of `applySfFxMiddleware` to ensure only one Secrets created at invoke time